File: PyProject/leetcode/link_table/lian-biao-zhong-dao-shu-di-kge-jie-dian-lcof.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        if not head:
            return None
        slow = head
        quick = head
        while quick:
            if k >= -1:
                k -= 1
                quick = quick.next
            else:
                quick = quick.next
                slow = slow.next
        return slow

File: GoProject/leetcode/linklist/lian-biao-zhong-dao-shu-di-kge-jie-dian-lcof.go
package linklist

func getKthFromEnd(head *ListNode, n int) *ListNode {
	first := head
	l := 0
	for head.Next != nil {
		l += 1
		head = head.Next
	}

	head = first
	if l < n {
		return head
	}

	m := l - n
	for j := 0; j < m; j++ {
		head = head.Next
	}
	head.Val = head.Next.Val
	head.Next = head.Next.Next

	return head

}
